### PR TITLE
Updates to allow retrieving reports and base report definitions

### PR DESCRIPTION
**Pull request by jamesshank**
## Before this PR

I need to pull Reports and Report Definitions to support needs.

## After this PR

We can now pull Reports and Report Definitions.

## Possible downsides?

Nope

### CODE REVIEW

File: tenablesc/report.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	BaseInfo
+	ReportDefinitionID string `json:"reportDefinitionID"`
+	JobID              string `json:"jobID"`
+	Type               string `json:"type"`
+	Status             string `json:"status"`
+	Running            string `json:"running"`
+	ErrorDetails       string `json:"errorDetails"`
+	TotalSteps         string `json:"totalSteps"`
+	CompletedSteps     string `json:"completedSteps"`
+	StartTime          string `json:"startTime"`
```
Comment (akarnani) [inline tenablesc/report.go]: I think the two time fields here could/should be `UnixEpochStringTime`

File: tenablesc/report.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	FinishTime         string `json:"finishTime"`
+	OwnerGID           string `json:"ownerGID"`
+	PubSites           []struct {
+		BaseInfo
+	} `json:"pubSites"`
+	Creator struct {
```
Comment (akarnani) [inline tenablesc/report.go]: Could this be `UserInfo`?

File: tenablesc/report.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		ID        string `json:"id"`
+		Username  string `json:"username"`
+		Firstname string `json:"firstname"`
+		Lastname  string `json:"lastname"`
+		UUID      string `json:"uuid"`
+	} `json:"creator"`
+	Owner struct {
```
Comment (akarnani) [inline tenablesc/report.go]: Could this be `UserInfo`?

**Comment by akarnani** [comment]
Hey @jamesshank,

Thanks for another contribution!  I had a few comments/questions on type choices for the new struct but otherwise looks great to me.

File: tenablesc/report.go (hunk 1 above)
Comment (jamesshank) [reply tenablesc/report.go]: Addressed in https://github.com/palantir/tenablesc-client/pull/28/commits/3a936f275c81fe1a175e00f915e76a11fce296ce

File: tenablesc/report.go (hunk 2 above)
Comment (jamesshank) [reply tenablesc/report.go]: Addressed in https://github.com/palantir/tenablesc-client/pull/28/commits/3a936f275c81fe1a175e00f915e76a11fce296ce

File: tenablesc/report.go (hunk 3 above)
Comment (jamesshank) [reply tenablesc/report.go]: Addressed in https://github.com/palantir/tenablesc-client/pull/28/commits/3a936f275c81fe1a175e00f915e76a11fce296ce

**Comment by jamesshank** [comment]
Thanks for the comments, @akarnani! I've addressed them all, nice catch. 

All the best!

**Comment by akarnani** [comment]
Thanks for the updates.  It looks like `uuid` is a core attribute of a `User` so perhaps we should lift it into `UserInfo`?

**Comment by jamesshank** [comment]
> Thanks for the updates. It looks like `uuid` is a core attribute of a `User` so perhaps we should lift it into `UserInfo`?

Per [here](https://github.com/palantir/tenablesc-client/blob/develop/tenablesc/base.go#L20-L27) I'm not seeing UUID, only ID. Can you advise?

**Comment by jamesshank** [comment]
> > Thanks for the updates. It looks like `uuid` is a core attribute of a `User` so perhaps we should lift it into `UserInfo`?
> 
> Per [here](https://github.com/palantir/tenablesc-client/blob/develop/tenablesc/base.go#L20-L27) I'm not seeing UUID, only ID. Can you advise?

Yep, I understand now. Forgive me. Just a sec addressing now.

**Comment by jamesshank** [comment]
@akarnani Feel free to take a look when you can! I think we should be good now. 

Thanks!

File: tenablesc/report.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	CompletedSteps     string              `json:"completedSteps"`
+	StartTime          UnixEpochStringTime `json:"startTime"`
+	FinishTime         UnixEpochStringTime `json:"finishTime"`
+	OwnerGID           string              `json:"ownerGID"`
+	PubSites           []struct {
+		BaseInfo
+	} `json:"pubSites"`
+	Creator    UserInfo `json:"creator"`
+	Owner      UserInfo `json:"owner"`
+	OwnerGroup struct {
```
Comment (akarnani) [inline tenablesc/report.go]: Can this just be `BaseInfo` instead of a struct that nests it?

File: tenablesc/report.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	Type               string              `json:"type"`
+	Status             string              `json:"status"`
+	Running            string              `json:"running"`
+	ErrorDetails       string              `json:"errorDetails"`
+	TotalSteps         string              `json:"totalSteps"`
+	CompletedSteps     string              `json:"completedSteps"`
+	StartTime          UnixEpochStringTime `json:"startTime"`
+	FinishTime         UnixEpochStringTime `json:"finishTime"`
+	OwnerGID           string              `json:"ownerGID"`
+	PubSites           []struct {
```
Comment (akarnani) [inline tenablesc/report.go]: Can this just be `[]BaseInfo` instead of a struct that nests it?

**Review by akarnani — COMMENTED** [review]
sorry to do this piecemeal, I just noticed one more thing.

File: tenablesc/report.go (hunk 5 above)
Comment (jamesshank) [reply tenablesc/report.go]: addressed in https://github.com/palantir/tenablesc-client/pull/28/commits/04d199ed5fa57bf5cc1acd9344dba94387dadfb4

File: tenablesc/report.go (hunk 4 above)
Comment (jamesshank) [reply tenablesc/report.go]: addressed in https://github.com/palantir/tenablesc-client/pull/28/commits/04d199ed5fa57bf5cc1acd9344dba94387dadfb4

**Review by akarnani — APPROVED** [review]

**Comment by akarnani** [comment]
Thanks again @jamesshank, LGTM.

**Pull request merged**